card/mod/core/set/all/event_conditions.rb

```diff
@@ -8,30 +8,15 @@ def event_applies? event
 
 private
 
-# if changing type, old card has set modules from old type, so we create
-# a new card to determine whether events apply.
-# (note: cached condition card would ideally be cleared after all
-# conditions are reviewed)
-def condition_card
-  @condition_card ||=
-    if updating_type?
-      cc = Card.fetch id, skip_modules: true
-      cc.name = name
-      cc.type_id = type_id
-      cc.content = content
-      cc.include_set_modules
-    else
-      self
-    end
+# existing card is being changed in a way that alters its sets
+def updating_sets?
+  @action == :update && real? &&
+    (attribute_is_changing?("type_id") || attribute_is_changing?("name"))
```

Comment:
you could use `type_id_is_changing?` and `name_is_changing` insterad

Reply:
ok.